src/core.js

```diff
@@ -35,7 +35,7 @@ async function blockJsRequests (page) {
 async function pruneNonCriticalCssLauncher ({
   browser,
   url,
-  astRules,
+  ast,
```

Comment:
regarding my performance comments - this is the relevant line (for 1.) - passing this (bigger) ast object here slows down the initialisation of this function, probably during parameter serialisation. Being able to pass through something that is quicker to serialize (smaller) should resolve this (particular) performance regression.

Reply:
Yes, a tree become bigger. I think we should produce a selector list from AST (array of strings) and pass that list to Puppetteer to filter it. It avoid us from passing a big JSON between processes and evaluating CSSTree into browser context (CSSTree doesn't require in browser context, because all work with AST can be done outside of it).

Reply:
Good idea, that would improve performance even further! Are you interested in taking on such work, or should we look to close this PR first, and look at that work separately?